lib/services/tokens/token.dao.js

```diff
@@ -72,27 +69,31 @@ dao.find = function (tokenObj, options) {
         return true;
       });
 
-      tokenPromises = activeTokenIds.map((id) => {
-        return db.hgetallAsync(tokenNamespace.concat(':', id))
-          .then((token) => {
-            let isEqual;
-
-            if (!token) {
-              return Promise.reject(new Error());
-            }
+      const getTokenPromise = new Promise((resolve, reject) => {
+        activeTokenIds.reverse();
 
-            isEqual = Object.keys(tokenObj).every((key) => tokenObj[key] === token[key]);
-            return isEqual ? token : Promise.reject(new Error());
-          });
+        const next = () => {
```

Comment:
I believe you have correct assumption that typically there are not many active tokens, most likely 0-1. So this approach should work